src/controllers/dex/tokens/tokenService.ts

```diff
@@ -0,0 +1,80 @@
+import { sql } from 'drizzle-orm';
+
+import { GIT_TOKEN_LIST_URL } from 'config/constants';
+import { db } from 'database/client';
+import { tokens } from 'database/schema/tokens';
+import { networks } from 'loader/networks';
+import { getLastUsdPrice } from 'loader/price';
+import { logger } from 'utils/logger';
+
+async function fetchTokensByChain(chainId: number) {
+  const tokensUrl = `${GIT_TOKEN_LIST_URL}/${chainId}/tokens.json`;
+  try {
+    const response = await fetch(tokensUrl);
+
+    if (!response.ok) {
+      throw new Error(`Failed to fetch tokens for chainId ${chainId}: ${response.statusText}`);
+    }
+
+    return await response.json();
+  } catch (error) {
+    console.error(`Error fetching tokens for chainId ${chainId}:`, error);
+    return [];
+  }
+}
+
+async function insertTokensToDatabase() {
+  const chains = networks.listChains();
```

Comment:
Stop cronjob for being fired again if job is still running.

```suggestion
async function insertTokensToDatabase(ctx: CronJob) {
  ctx.stop();
  const chains = networks.listChains();
```